# Mac set-up for RADseq
Here you will find a list of usefull programs and R packages usefull for performing **RADseq studies**.

## BASH commands
### Get Xcode
```{r, engine = 'bash', eval = FALSE}
xcode-select --install
```

### Install Homebrew
```{r, engine = 'bash', eval = FALSE}
/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
```

### Install GDAL
```{r, engine = 'bash', eval = FALSE}
brew install gdal --HEAD
```

### Install SHUFF command
```{r, engine = 'bash', eval = FALSE}
brew install coreutils
```

### Install wget
```{r, engine = 'bash', eval = FALSE}
brew install wget
```

### Install gsl
```{r, engine = 'bash', eval = FALSE}
brew install gsl
```

## Programs
### Install FILEZILLA
```{r, engine = 'bash', eval = FALSE}
ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)" < /dev/null 2> /dev/null ; brew install caskroom/cask/brew-cask 2> /dev/null
brew cask install filezilla
```

## Install [VCFTOOLS](http://vcftools.sourceforge.net/)
VCFtools is a program package designed for working with VCF files, such as those generated by [STACKS](http://catchenlab.life.illinois.edu/stacks/).
```{r, engine = 'bash', eval = FALSE}
brew install vcftools
```

## Install [PLINK](http://zzz.bwh.harvard.edu/plink/download.shtml)
PLINK is a free, open-source whole genome association analysis toolset, designed to perform with tfam and tped files that are easily converted from vcftools via the ***plink-tped*** following command:
```{r, engine = 'bash', eval = FALSE}
vcftools --vcf nameofyourfile.vcf --plink-tped --out nameofyourfile
done
```

### Install JAVA
```{r, engine = 'bash', eval = FALSE}
brew cask install java
brew cask info java
```

### Install [PGDSpider](http://www.cmpg.unibe.ch/software/PGDSpider/)
PGDSpider is a data conversion tool for population genetic and genomics programs. 
It support a vast range of data types with the ability of parsing 33 and writing 36 different file formats


## Install [ADMIXTURE](http://software.genetics.ucla.edu/admixture/)
ADMIXTURE is a software tool for maximum likelihood estimation of individual ancestries from multilocus SNP genotype datasets. It uses the same statistical model as STRUCTURE but calculates estimates much more rapidly using a fast numerical optimization algorithm.
To use ADMIXTURE, you need to transfor your file vcf file into a bed format via plink by typing the ***make-bed*** command available in PLINK:
```{r, engine = 'bash', eval = FALSE}
plink --tfam nameofyourfile.tfam --tped nameofyourfile.tped --make-bed --out nameofyourfile
done
```

### Install BEDOPS
```{r, engine = 'bash', eval = FALSE}
git clone https://github.com/bedops/bedops.git
cd bedops
make
make install
```

### Install [R studio](https://www.rstudio.com/products/rstudio/download/)
R studio has developped an integrated development environment for R, with a console that supports direct code execution.

```{r, engine = 'bash', eval = FALSE}
brew cask install xquartz
brew cask install r
brew cask install --appdir=/Applications rstudio
```

## R software
# Install R package using devtools package
First install devtools package that allows you to directly download R package available in github.
```{r}
install.packages("devtools")
```

Then, a variety of useful package for population genomics such as ***adegenet***, ***strataG***, ***assigner*** are available in R.

```{r}
install.packages("adegenet")
install.packages("strataG")
install.packages("assigner")
install.packages("radiator")
install.packages("ape")
install.packages("poppr")
install.packages("hierfstat")
install.packages("dartR")
install.packages("StAMPP")
install.packages("genetics")
install.packages("LDna")
install.packages("ade4")
devtools::install_github('royfrancis/pophelper')
devtools::install_github("thierrygosselin/assigner")
```

### Set up your bash profile in MAcOs Catalina

Forst, create a folder named 00-Programs and placed it in your local environment, with a short cut access. 
In this folder you will enter all the binaies that you need to use for running programs such as **admixture**, **Bayescan** or **VCFTOOLS** for instance.

Create your bash_profile that is now named zprofile in macOS Catalina.
```{r, engine = 'bash', eval = FALSE}
nano .zprofile
```

Add this folder path to your zprofile file.
```{r, engine = 'bash', eval = FALSE}
export PATH=$PATH:/Users/laurabenestan/Dropbox/02-Bio-informatics/00-Programs
```
